Classes/Console/Command/ConfigurationCommandController.php

```diff
@@ -159,15 +159,22 @@ public function showLocalCommand($path, $json = false)
      *
      * @param string $path Path to system configuration
      * @param string $value Value for system configuration
+     * @param string $type Datatype of the configuration value
      */
-    public function setCommand($path, $value)
+    public function setCommand($path, $value, $type = '')
```

Comment:
Another option would be to add a `--as-json` flag (`$asJson = false`)
and do a `\json_decode($value, true)`

This would allow `typo3cms configuration:set SYS/caching/cacheConfigurations/cache_hash/options/database 2 --as-json` to set an integer value. Another benefit would be that it would allow setting arrays as values.

What do you think?

Reply:
This is a really good idea! With this flag there is no need to define the datatype.